ttnn/cpp/ttnn/operations/data_movement/slice/slice.cpp

```diff
@@ -188,109 +178,100 @@ template<>
 ttnn::Tensor SliceOperation::invoke<uint32_t, 4>(
     uint8_t queue_id,
     const ttnn::Tensor& input_tensor,
-    const std::array<uint32_t, 4> &output_tensor_start,
-    const std::array<uint32_t, 4> &output_tensor_end,
+    const std::array<uint32_t, 4> &begins,
+    const std::array<uint32_t, 4> &ends,
     const std::array<uint32_t, 4> &step,
     const std::optional<MemoryConfig>& memory_config_arg,
     const std::optional<Tensor>& optional_output_tensor) {
 
-    bool efficient_path = input_tensor.storage_type() == StorageType::DEVICE && !input_tensor.is_sharded();
-    for (int i = 0; i < 4; ++i) {
-        efficient_path &= (step[i] == 1) && (output_tensor_end[i] > output_tensor_start[i]);
-        if (!efficient_path) {
-            break;
+    const auto& padded_input_shape = input_tensor.get_shape().with_tile_padding();
+    TT_FATAL(padded_input_shape.rank() == 4, "Input tensor must have rank 4");
+
+    bool no_step = step[0] == 1 && step[1] == 1 && step[2] == 1 && step[3] == 1;
+    bool starts_zero = begins[0]==0 && begins[1]==0 && begins[2]==0 && begins[3]==0;
+    bool ends_max = ends[0]==padded_input_shape[0] && ends[1]==padded_input_shape[1] && ends[2]==padded_input_shape[2] && ends[3]==padded_input_shape[3];
+
+    if (no_step && starts_zero && ends_max) {
+        if (input_tensor.storage_type() == StorageType::DEVICE) {
+            auto memory_config = optional_output_tensor.has_value() ? optional_output_tensor.value().memory_config() : memory_config_arg.value_or(input_tensor.memory_config());
+            return ttnn::to_memory_config(input_tensor, memory_config, std::nullopt);
         }
-    }
-    if (efficient_path) {
-        return operation::run(
-            SliceDeviceOperation{
-                tt::tt_metal::LegacyShape(output_tensor_start),
-                tt::tt_metal::LegacyShape(output_tensor_end),
-                std::nullopt,
-                memory_config_arg.value_or(input_tensor.memory_config())},
-            {input_tensor}, {}, {optional_output_tensor}, queue_id)
-            .at(0);
+        return input_tensor;
     }
 
-    const auto& input_shape = input_tensor.get_legacy_shape();
-    std::array<uint32_t, 4> modified_begins;
-    std::array<uint32_t, 4> modified_ends;
-    std::array<uint32_t, 4> modified_step = step;
+    const bool tiled = input_tensor.get_layout() == Layout::TILE;
+    bool on_device = input_tensor.storage_type() == StorageType::DEVICE;
 
-    bool no_step = true;
-    bool empty = false;
     std::array<uint32_t, 4> actual_shape;
     std::array<uint32_t, 4> padded_shape;
```

Comment:
Can also use `ttnn::SimpleShape` for args like this in the future. This is also fine.